### DISCUSSION TITLE
Should Gatekeeper follow a factory pattern

### DISCUSSION THREAD

**Discussion by thelostone-mc** [gitcoinco/grants-stack#954] [Q&A]
### Description

Is there a reason for the gatekeeper to follow a factory pattern like the program , round, votingStrategy factory contracts ?
The only benefit I can see if that we can reduce the gas costs cause we'd have an implementation contract which can be cloned as opposed to deploying a fresh contract every time 

This is the only added benefit I can think of ATM

**Comment by thelostone-mc** [comment]
Thinking out loud -> Should we do the same for the payout strategy just to reduce gas costs

**Comment by chibie** [comment]
+1 on doing the same factory pattern.

for the above reason @thelostone-mc gave, have a sort of log of all deployed contracts with same template, and also as a consistent/default pattern for all our  contracts now and in the future.

**Comment by vacekj** [comment]
Closing as stale.
